pkg/provider/kubernetes/gateway/kubernetes.go

```diff
@@ -492,21 +492,28 @@ func (p *Provider) fillGatewayConf(ctx context.Context, client Client, gateway *
 }
 
 func gatewayHTTPRouteToHTTPConf(ctx context.Context, ep string, listener v1alpha1.Listener, gateway *v1alpha1.Gateway, client Client, conf *dynamic.Configuration) []metav1.Condition {
-	// TODO: support RouteNamespaces
 	selector := labels.Everything()
 	if listener.Routes.Selector != nil {
 		selector = labels.SelectorFromSet(listener.Routes.Selector.MatchLabels)
 	}
 
-	httpRoutes, err := client.GetHTTPRoutes(gateway.Namespace, selector)
+	namespace := gateway.Namespace
+	if listener.Routes.Namespaces != nil && listener.Routes.Namespaces.From != nil {
+		// TODO: Implement v1alpha1.RouteSelectSelector
+		if *listener.Routes.Namespaces.From == v1alpha1.RouteSelectAll {
+			namespace = metav1.NamespaceAll
+		}
+	}
```

Comment:
Should this be in a method since it is re-used? This would also allow `v1alpha1.RouteSelectSelector` to be implemented in one place, rather than replicated.

Reply:
Solved with https://github.com/traefik/traefik/pull/8299/commits/fa127abb8730d2d17b751a188c55c9fb9c9f4a90 :)